transformers/senml/transformer.go

```diff
@@ -59,9 +57,10 @@ func (t transformer) Transform(msg messaging.Message) (interface{}, error) {
 	msgs := make([]Message, len(normalized.Records))
 	for i, v := range normalized.Records {
 		// Use reception timestamp if SenML messsage Time is missing
-		t := v.Time
-		if t == 0 {
-			t = float64(time.Now().UnixNano())
+		time := v.Time
+		if time == 0 {
+			// Convert the timestamp into float64 with nanoseconds precision
+			time = float64(msg.Created.GetSeconds()) + float64(msg.Created.GetNanos())/float64(1e9)
```

Comment:
This looks like precision in seconds to me. If you wanted nanoseconds, you should probably do something like:
```go
float64(msg.Created.GetSeconds() * 1e9 + msg.Created.GetNanos())
```